monkey/infection_monkey/network/network_scanner.py

```diff
@@ -62,12 +57,59 @@ def _get_inaccessible_subnets_ips(self):
 
         return subnets_to_scan
 
-    def get_victim_machines(self, scan_type, max_find=5, stop_callback=None):
+    @staticmethod
+    def get_k8s_node_subnet(system_info):
+        """
+        Gets suspected subnet of k8s nodes.
+        Second hop of traceroute out of a pod is usually in class C with k8s nodes.
+        :param system_info: system_info returned from system_info_collector.
+        :return: Suspected subnet of k8s nodes or None if machine is not a k8s pod.
+        """
+        if system_info and system_info.get('k8s', {'is_pod': False})['is_pod']:
+            route = traceroute('google.com', 2)
+            if len(route) >= 2:
+                return CidrRange(str(route[-1]) + '/24')
+
+        return None
+
+    def generate_ranges(self, system_info):
+        """
+        Generates network ranges to scan. Re-queries the island's config after iterating over all ranges in local config
+        :return: yields network range
+        """
+        old_range_strs = set()
+        range_strs = set()
+
+        while True:
+            old_range_strs = old_range_strs.union(range_strs)
+            ControlClient.load_control_config()
+            range_strs = set(WormConfiguration.subnet_scan_list + WormConfiguration.dynamic_subnet_scan_list)
+            range_strs = range_strs.difference(old_range_strs)
+
+            if not range_strs:
+                break
+
+            for range_str in range_strs:
+                yield NetworkRange.get_range_obj(address_str=range_str)
+
+        if WormConfiguration.local_network_scan:
+            for net_range in get_interfaces_ranges():
+                yield net_range
+
+        k8s_node_subnet = NetworkScanner.get_k8s_node_subnet(system_info)
+        if k8s_node_subnet:
+            yield k8s_node_subnet
+
+        for net_range in self._get_inaccessible_subnets_ips():
+            yield net_range
+
+    def get_victim_machines(self, scan_type, max_find=5, stop_callback=None, system_info=None):
         """
         Finds machines according to the ranges specified in the object
         :param scan_type: A hostscanner class, will be instanced and used to scan for new machines
         :param max_find: Max number of victims to find regardless of ranges
         :param stop_callback: A callback to check at any point if we should stop scanning
+        :param system_info: system_info returned from system_info_collector
```

Comment:
I don't understand why this is here.
We collect system info in `monkey.py` before we initialize the network scanner. We can pass it there.